### PR TITLE
TechDraw: Ext. Tools: Improve menu texts and tooltips

### PR DESCRIPTION
**Pull request by Roy-043**
PR to make the GUI texts of the extension tools more consistent.

* Title case for all menu texts. This is consistent with most TechDraw menus.
* Linebreaks in tooltips should be `<br>`.
* Displaying the text of multiline tooltips in the statusbar does not work. So the menu text is now used for the statusbar text.

Additionally:
The order of the tools in the menu did not match the order in the toolbar. This was corrected.

Forum topic:
https://forum.freecadweb.org/viewtopic.php?f=35&t=65436

- [x]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [x]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [x]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the [FreeCAD bug tracker issue number](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) in case a particular commit solves or is related to an existing issue on the tracker. Ex: `Draft: fix typos - fixes #0004805`

---

### CODE REVIEW

**Comment by Roy-043** [comment]
I do no know CPP, but while working on the files I noticed what I think is an inconsistency:

In the `CmdTechDrawExtensionPositionSectionView` code the 'empty selection' check is different `_checkSel` is not used:
https://github.com/FreeCAD/FreeCAD/blob/59c470dedf28da2632abfe6e4481ce09aaf6e233/src/Mod/TechDraw/Gui/CommandExtensionPack.cpp#L1369-L1379

@edi271 Can you check this? Thanks.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5425">pipeline status</a> for feature branch [PR_5425](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5425). Pipeline [454285311](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/454285311) was triggered at [1b1da8e](https://github.com/FreeCAD/FreeCAD/pull/5425/commits/1b1da8e4d9d8fcc62aabf0eb0d31e3d1658dd967). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/TechDraw/Gui/CommandExtensionDims.cpp [hunk 1]
```diff
@@ -141,15 +141,15 @@ DEF_STD_CMD_A(CmdTechDrawExtensionInsertDiameter)
...
-    - click this button");
-    sWhatsThis = "TechDraw_ExtensionInsertDiameter";
-    sStatusTip = sToolTipText;
-    sPixmap = "TechDraw_ExtensionInsertDiameter";
+    sAppModule      = "TechDraw";
+    sGroup          = QT_TR_NOOP("TechDraw");
+    sMenuText       = QT_TR_NOOP("Insert '⌀' Prefix");
+    sToolTipText    = QT_TR_NOOP("Insert a '⌀' symbol at the beginning of the dimension text:<br>\
+- Select one or more dimensions<br>\
+- Click this tool");
```
Comment (luzpaz) [inline src/Mod/TechDraw/Gui/CommandExtensionDims.cpp:149]: A newline should be enough. No need to add `<br>` (also it confuses the translators)

File: src/Mod/TechDraw/Gui/CommandExtensionDims.cpp [hunk 2]
```diff
@@ -1864,7 +1894,7 @@ void execCreateVertChamferDimension(Gui::Command* cmd) {
     TechDraw::DrawViewPart* objFeat;
     if (!_checkSelAndObj(cmd, selection, objFeat, "TechDraw Create Vertical Chamfer Dimension"))
         return;
-    Gui::Command::openCommand(QT_TRANSLATE_NOOP("Command", "Vert Chamfer Dim"));
+    Gui::Command::openCommand(QT_TRANSLATE_NOOP("Command", "Create Vert Chamfer Dim"));
```
Comment (luzpaz) [inline src/Mod/TechDraw/Gui/CommandExtensionDims.cpp:1897]: Wouldn't it make sense to spell out `Vertical` fully here?

**Review by luzpaz — CHANGES_REQUESTED** [review]

File: src/Mod/TechDraw/Gui/CommandExtensionDims.cpp (hunk 1 above)
Comment (Roy-043) [reply src/Mod/TechDraw/Gui/CommandExtensionDims.cpp:149]: With the new tooltip HTML formatting `<br>` is required.
Forum topic:
https://forum.freecadweb.org/viewtopic.php?t=62587
Image:
tooltip-no-linebreaks

File: src/Mod/TechDraw/Gui/CommandExtensionDims.cpp (hunk 2 above)
Comment (Roy-043) [reply src/Mod/TechDraw/Gui/CommandExtensionDims.cpp:1897]: AFAIK this text is used to identify commits. The user will only be able to see this when using the dropdown arrows next to Undo and Redo. In this sense it is not very important.

My reason for adding "Create" is that the command is called `TechDraw_ExtensionCreateVertChamferDimension`.

I have mainly focused on menu texts and tooltips. This was already an involved process that took a lot more time than anticipated. I was tempted to do more, I particularly don't like some of the command names, but you have to draw the line somewhere...

File: src/Mod/TechDraw/Gui/CommandExtensionDims.cpp (hunk 1 above)
Comment (luzpaz) [reply src/Mod/TechDraw/Gui/CommandExtensionDims.cpp:149]: Are you kidding?!? :facepalm: that means _so_ many translations will need to be redone!

File: src/Mod/TechDraw/Gui/CommandExtensionDims.cpp (hunk 2 above)
Comment (luzpaz) [reply src/Mod/TechDraw/Gui/CommandExtensionDims.cpp:1897]: Ok, fair enough

File: src/Mod/TechDraw/Gui/CommandExtensionDims.cpp (hunk 1 above)
Comment (Roy-043) [reply src/Mod/TechDraw/Gui/CommandExtensionDims.cpp:149]: I do not know how many tooltips contain linebreaks, but it is not that common.

**Comment by Roy-043** [comment]
Note that `TechDraw_ArchView` is missing in the `CommandBars` section in `Workbench.cpp`.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5425">pipeline status</a> for feature branch [PR_5425](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5425). Pipeline [454557981](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/454557981) was triggered at [138cc93](https://github.com/FreeCAD/FreeCAD/pull/5425/commits/138cc939d709bd238e3a7f11f3d408182d02b3cb). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by edi271** [comment]
> I do no know CPP, but while working on the files I noticed what I think is an inconsistency:
> 
> In the `CmdTechDrawExtensionPositionSectionView` code the 'empty selection' check is different, `_checkSel` is not used:
> 
> https://github.com/FreeCAD/FreeCAD/blob/59c470dedf28da2632abfe6e4481ce09aaf6e233/src/Mod/TechDraw/Gui/CommandExtensionPack.cpp#L1369-L1379
> 
> @edi271 Can you check this? Thanks.

The existing code is OK.
In most tools an object (cosmetic line, dimension etc.) must be selected.
_checkSel executes two checks:
1. check if a valid "container" (= view) is selected. If failed: "Selection is empty"
2. if check 1 passed, check if an object is selected. If failed: "No object selected"

In case of PositionSectionView only the "container" is used, check 2 always would fail in _checkSel

**Review by edi271 — APPROVED** [review]
File CommanExtensionDims:
The ...CoordDimension commands (horizontal/vertical/oblique) have been extended in PR 5395 Now the baseline can be positioned left or right.
Unfortunately I did not update the toolTip dialog. I apologise. There should be added lines like:

- If the first vertex is left of the second one, the baseline is positioned left.
- If the first vertex is right of the second one, the baseline is positioned right.

File CommanExtensionPack:
Similar issue in the commands ...ExtendLine and ...ShortenLine
There should be added a line:

- Select the extension value (optional)

**Comment by Roy-043** [comment]
> In case of PositionSectionView only the "container" is used, check 2 always would fail in _checkSel
Thanks for explaining.

I have updated the tooltips bases on you suggestions.

**Comment by Roy-043** [comment]
There is a "pipeline failed" message. I do not know if, and how, this is related to this PR.

**Comment by edi271** [comment]
> There is a "pipeline failed" message. I do not know if, and how, this is related to this PR.

I had this problem in a past PR. The cause was a syntax error stopping the compilation.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5425">pipeline status</a> for feature branch [PR_5425](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5425). Pipeline [455231818](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/455231818) was triggered at [b672ef7](https://github.com/FreeCAD/FreeCAD/pull/5425/commits/b672ef70febcb9ad5b6756ebf7074b6225bd69ef). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by Roy-043** [comment]
I think I have found and fixed the cause of the "pipeline failed" message: missing closing bracket.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5425">pipeline status</a> for feature branch [PR_5425](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5425). Pipeline [455773683](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/455773683) was triggered at [c1c2253](https://github.com/FreeCAD/FreeCAD/pull/5425/commits/c1c225325726fff1f8da3ffeb5d6836ba4cd7122). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by donovaly** [comment]
@WandererFan , could you please squash the commits into one when merging? (of course only if the different commits should not be kept).
This helps a lot when e.g. backporting PRs. (I don't want to backport this PR, just a general note.)

**Comment by WandererFan** [comment]
I don't know how to do this in GitHub.  Is there an equivalent to "rebase
-i"?

**Comment by donovaly** [comment]
> I don't know how to do this in GitHub.

Here is a PR with 7 commits: https://github.com/FreeCAD/FreeCAD/pull/5442
- to squash them into one when merging, click on "Squash and merge":
firefox_21hZzB44xZ

To keep all commits, use "Rebase and merge".
For PRs that already only have one commit "Rebase and merge" is also the preferred method because it keeps the info who was the author and who only performed the merge.

**Pull request merged**